### ISSUE TITLE
Sync: HPOS orders sync refunds incorrectly.

### ISSUE THREAD

**Issue by ibndawood** [Automattic/jetpack#39068]
### Impacted plugin

Jetpack, Backup

### Quick summary

When a post of `shop_order_refund` is synced, the `total_amount` column value is always stored as `0.00` in the `wc_orders` table. This needs to be corrected. The expected `total_amount` value should be `-1 * refund_amount`. The `refund_amount` value is synced but is not stored.

### Steps to reproduce

1. Select an order that is synced already and has a status `Pending Payment` or `On hold`. 
2. If such an order does not exist, please create a new order and verify it is synced. 
3. Let's assume the `total_amount` of the order is `50.0000`.
4. In the WordPress admin panel, click on the order to edit.
5. In the Order Edit screen, click the `Status` dropdown and select `Refunded` to refund the order entirely.
6. In the local database, verify a new record is created with `parent_order_id` as the ID of the order refunded.
7. Verify the `total_amount` column in the local database equals `*1 * total_amount of the parent_order_id`. i.e. `-50.0000`.
8. Check the record in the synced WPCOM table. 
9. The `total_amount` is set to `0.00`.

### A clear and concise description of what you expected to happen.

The `total_amount` should not be zero and should be equal to the total amount of the parent order in negative.

### What actually happened

The `total_amount` is zero.

### Impact

All

### Available workarounds?

Yes, easy to implement

### If the above answer is "Yes...", outline the workaround.

We can use the `jetpack_sync_filtered_hpos_order_data` filter to set `order_total` index which is saved as the `total_amount`.

**Comment by fgiannar** [comment]
I don't think this is related to Sync. It looks like when the `woocommerce_after_order_refund_object_save` action is fired (aka **after** an order refund is saved) the `total_amount` is 0, while it should be `-1 * refund_amount` as mentioned in the description of this issue.

Related discussion: p9dueE-8ec-p2#comment-10528

@barryhughes Do you think this report is sufficient to take a look at this and confirm if it's indeed a Woo bug? Is it possible that when the `woocommerce_after_order_refund_object_save` action is fired, the corresponding `WC_Order` object for the refund is not yet updated with the expected total?

**Comment by barryhughes** [comment]
> It looks like when the `woocommerce_after_order_refund_object_save` action is fired (aka after an order refund is saved) the total_amount is 0.

Yes ✅ 

The dynamic hook `woocommerce_after_<TYPE>_object_save` fires immediately after the `save` method is called on a `WC_Data` instance. It may in fact be called multiple times as different pieces of information are calculated and persisted. 

> While it should be -1 * refund_amount as mentioned in the description of this issue.

I think this can only be expected once the refund is fully established, and the hook you are using doesn't offer that guarantee: it will be triggered multiple times during the creation of a refund, including before the point when the total is determined.

Can you use `woocommerce_create_refund` instead?

**Comment by fgiannar** [comment]
A status update here, since the issue was marked as stale. @Automattic/jetpack-vulcan team plan to work on this issue as part of a Spike, where we intend to investigate a more generic approach (not limited to refunds) for syncing WooCommerce HPOS orders. 
The idea is to replace the "costly" `woocommerce_after_order_object_save` actions (that get triggered quite a lot) with relevant less noisy ones.
Noting that we will probably need to collaborate with a WooCommerce team, potentially @Automattic/ventures (working on the WooCommerce Analytics plugin) to ensure we don't miss any data currently needed (eg draft orders).
